sources/Component/Collision/CollisionComponent.hpp

```diff
@@ -0,0 +1,55 @@
+//
+// Created by Louis Auzuret on 2021-05-20.
+//
+
+#pragma once
+
+#include "Models/Callback.hpp"
+#include "Models/Vector3.hpp"
+#include "Component/Component.hpp"
+#include "Entity/Entity.hpp"
+
+namespace BBM 
+{
+	class CollisionComponent : public WAL::Component
+	{
+		private:
+		public:
+			//! @brief Bound size on all axis
+			Vector3f bound;
+			//! @brief onCollide functions to be called
+			WAL::Callback<WAL::Entity &, const WAL::Entity &> onCollide;
```

Comment:
You could add a onCollided callback called on the update of the other entity like that:
```c++
WAL::Callback<const Wal::Entity &, WAL::Entity &> onCollided;
```

This would allow walls and other colliders to handle what they do to players and so on.
If we don't have this, the onCollide of the player will have every collision interactions (walls, holes, items...)